### PR TITLE
feat: append generated router key to world.toml

### PR DESCRIPTION
**Pull request by zulkhair**
Closes: WORLD-990

## Overview

World CLI generates router key and append it to world.toml file

## Brief Changelog

- create function for appending the section and field to toml file
- create function for generating random hex string

## Testing and Verifying

- create unit test for added function
- test the env by running command `world cardinal start`

### CODE REVIEW

**Comment by zulkhair** [comment]
* **#51** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/51?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @zulkhair and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: common/teacmd/git.go [hunk 1]
```diff
@@ -132,3 +157,55 @@ func replaceInFile(filePath, oldStr, newStr string) error {
...
+// appendToToml reads the given TOML file, appends the specified section and fields, and writes it back.
+func appendToToml(filePath, section string, fields map[string]interface{}) error {
+	// Read the existing content of the TOML file
+	fileContent, err := os.ReadFile(filePath)
+	if err != nil {
+		return eris.Wrap(err, "error reading file")
+	}
+
+	// Unmarshal the file content into a map
+	var config map[string]interface{}
```
Comment (technicallyty) [inline common/teacmd/git.go:170]: nit: you can actually use `any` in place of `interface{}` and it looks very pretty in my opinion 😛

File: common/teacmd/git_test.go [hunk 2]
```diff
@@ -72,3 +73,96 @@ func cleanUpDir(targetDir string) {
...
+			},
+			expectedError: nil,
+		},
+		{
+			name:     "create new section and append fields",
+			filePath: tempFile.Name(),
+			section:  "new_section",
+			fields: map[string]interface{}{
+				"field3": true,
+				"field4": 3.14,
```
Comment (technicallyty) [inline common/teacmd/git_test.go:119]: now i want dessert..

File: common/teacmd/git_test.go [hunk 3]
```diff
@@ -72,3 +73,96 @@ func cleanUpDir(targetDir string) {
...
 	}
 }
+
+func TestAppendToToml(t *testing.T) {
+	// Create a temporary TOML file for testing
+	tempFile, err := os.CreateTemp("", "test.toml")
+	if err != nil {
+		t.Fatalf("failed to create temporary file: %v", err)
+	}
+	defer os.Remove(tempFile.Name())
```
Comment (technicallyty) [inline common/teacmd/git_test.go:83]: you can also use `t.Cleanup`, which might be safer in cases where we use t.Parallel() for parallelizing our tests.

File: common/teacmd/git.go [hunk 4]
```diff
@@ -3,11 +3,27 @@ package teacmd
...
+)
+
+const (
+	oldModuleName = "github.com/argus-labs/starter-game-template/cardinal"
+
+	tomlFile = "world.toml"
+
+	routerKey = "ROUTER_KEY"
+
+	tomlSection = "common"
```
Comment (technicallyty) [inline common/teacmd/git.go:24]: this should prob be a bit more specific like `tomlSectionCommon`

File: common/teacmd/git.go [hunk 5]
```diff
@@ -3,11 +3,27 @@ package teacmd
...
+
+	keyLength = 32
```
Comment (technicallyty) [inline common/teacmd/git.go:26]: same here, maybe `routerKeyLength`?

File: common/teacmd/git.go (hunk 1 above)
Comment (zulkhair) [reply common/teacmd/git.go]: updated

File: common/teacmd/git.go (hunk 4 above)
Comment (zulkhair) [reply common/teacmd/git.go]: updated

File: common/teacmd/git.go (hunk 5 above)
Comment (zulkhair) [reply common/teacmd/git.go]: updated

File: common/teacmd/git_test.go (hunk 3 above)
Comment (zulkhair) [reply common/teacmd/git_test.go]: updated using `t.Cleanup`

File: common/teacmd/git.go [hunk 6]
```diff
@@ -58,12 +74,21 @@ func GitCloneCmd(url string, targetDir string, initMsg string) error {
 		return err
 	}
 
-	oldModuleName := "github.com/argus-labs/starter-game-template/cardinal"
 	err = refactorModuleName(oldModuleName, filepath.Base(targetDir))
 	if err != nil {
 		return err
 	}
 
+	randomString, err := generateRandomHexString(routerKeyLength)
```
Comment (technicallyty) [inline common/teacmd/git.go:82]: i would call this `rtrKey` to be specific

**Review by technicallyty — APPROVED** [review]
LGTM, one last nit !

**Pull request merged**